File: Backend/QuanLyKhamBenhAPI/Models/ChatMessage.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuanLyKhamBenhAPI.Models
{
    public class ChatMessage
    {
        [Key]
        public int MessageId { get; set; }

        [Required]
        public int PatientId { get; set; }

        public int? AdminId { get; set; }

        [Required]
        [MaxLength(2000)]
        public string Message { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        public string SenderRole { get; set; } = string.Empty; // "Patient" hoặc "Admin"

        public DateTime SentAt { get; set; } = DateTime.UtcNow;

        public bool IsRead { get; set; } = false;

        // Navigation properties
        [ForeignKey(nameof(PatientId))]
        public virtual UserAccount? Patient { get; set; }

        [ForeignKey(nameof(AdminId))]
        public virtual UserAccount? Admin { get; set; }
    }
}

File: FrontendMobile/lib/models/chat_message.dart
class ChatMessage {
  final int? messageId;
  final String text;
  final bool isUser;
  final DateTime timestamp;
  final String? senderRole; // 'Patient', 'Admin', 'AI'
  final String? senderName;

  ChatMessage({
    this.messageId,
    required this.text,
    required this.isUser,
    required this.timestamp,
    this.senderRole,
    this.senderName,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      messageId: json['messageId'],
      text: json['message'],
      isUser: json['senderRole'] == 'Patient',
      timestamp: DateTime.parse(json['sentAt']),
      senderRole: json['senderRole'],
      senderName: json['adminName'],
    );
  }
}
